src/beman/optional26/tests/detail/iterator.t.cpp

```diff
@@ -10,18 +10,27 @@
 #include <tests/test_types.hpp>
 
 #include <concepts>
+#include <cstdlib>
 #include <type_traits>
 #include <set>
 #include <vector>
 
-namespace beman::optional26::test {
+#define CONSTEXPR_EXPECT_EQ(val1, val2)   \
```

Comment:
I don't understand why this CONSTEXPR_ set of functions was introduced. Should we use `static_assert` be used to check the constexpr functionality?

Reply:
I'm already using `static_assert((lambda(), true));` now. These set of wrapping macros were introduced because gtest's corresponding macros are not constexpr-friendly.

Reply:
First of all, this seems to be duplicated code from src/beman/optional26/tests/optional_range_support.t.cpp. So if we'll keep it, @frederick-vs-ja please move it in `est_utilities/test_types.*`.

2nd observation: I think we want to have all constexpr tests in a separate file, similar to https://github.com/beman-project/optional26/blob/main/src/beman/optional26/tests/optional_constexpr.t.cpp (optional.t.cpp vs optional_constexpr.t.cpp). @camio , @steve-downey ?

3rd: I think static_assert should work. IF not, this is how we do constexpr testing right now - https://github.com/beman-project/optional26/blob/main/src/beman/optional26/tests/optional_constexpr.t.cpp#L251-L263. Hope it hepls.

Reply:
> I'm already using static_assert((lambda(), true)); now. These set of wrapping macros were introduced because gtest's corresponding macros are not constexpr-friendly.

Oh, I see. You're doing this so you can use something like EXPECT_EQ from within a lambda evaluated at compile time. Cool.